litmus-portal/backend/auth/pkg/server/server.go

```diff
@@ -31,15 +45,77 @@ func NewServer(cfg *Config) *Server {
 	srv := &Server{
 		Config:  cfg,
 		Manager: manager,
-	}
+		GithubConfig: oauth2.Config{
+			ClientID:     os.Getenv("ClientID"),
+			ClientSecret: os.Getenv("ClientSecret"),
+			Scopes:       []string{"read:user", "user:email"},
+			RedirectURL:  "http://localhost:3000/oauth/github",
+			Endpoint:     githubAuth.Endpoint,
+		}}
 
 	return srv
 }
 
-// Server Provide authorization server
-type Server struct {
-	Config  *Config
-	Manager *manage.Manager
+//Middleware redirects to a github endpoint to get the temp code for oauth
+func (s *Server) Middleware(c *gin.Context) {
+
+	//log.Println(s.config.ClientID)
+	var w http.ResponseWriter = c.Writer
+	var r *http.Request = c.Request
+	u := s.GithubConfig.AuthCodeURL("xyz")
+	http.Redirect(w, r, u, http.StatusFound)
+}
+
+//GitHub gets the temp code for oauth and exchanges this code with github in order to get auth token
+func (s *Server) GitHub(c *gin.Context) {
+	var w http.ResponseWriter = c.Writer
+	var r *http.Request = c.Request
+	r.ParseForm()
+	state := r.Form.Get("state")
+
+	if state != "xyz" {
+		http.Error(w, "State invalid", http.StatusBadRequest)
+		return
+	}
+	code := r.Form.Get("code")
+	if code == "" {
+		http.Error(w, "Code not found", http.StatusBadRequest)
+		return
+	}
+	token, err := s.GithubConfig.Exchange(context.Background(), code)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	userGDetails.OAuthToken = *token
+
```

Comment:
try to avoid newlines wherever possible for better readablility of the code. For better BCH Score we can have at max 15lines in a go without a single new line. So just keep that in mind